src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/MetricsStore.cs

```diff
@@ -153,14 +161,17 @@ public async Task SnapshotMetrics(Stream outputStream, CancellationToken token)
         private static string GetMetricLabels(ICounterPayload metric, double? quantile)
         {
             string metadata = metric.Metadata;
+
+            char separator = IsMeter(metric) ? '=' : ':';
+            var metadataValues = CounterUtilities.GetMetadata(metadata, separator);
             if (quantile.HasValue)
             {
-                metadata = CounterUtilities.AppendPercentile(metadata, quantile.Value);
+                metadataValues.Add("quantile", quantile.Value.ToString(CultureInfo.InvariantCulture));
```

Comment:
In converting from percentile to quantile, do we need to divide by 100? E.g. 95% is 0.95 quantile I think? Or is that already getting taken care of

Reply:
quantiles all start as values between 0 and 1. AppendPercentile converts them to percentages.